auraed/src/vms/virtual_machines.rs

```diff
@@ -0,0 +1,129 @@
+use std::{collections::HashMap, net::Ipv4Addr};
+
+use anyhow::anyhow;
+use net_util::MacAddr;
+use vmm_sys_util::{rand, signal::block_signal};
+
+use super::virtual_machine::{NetSpec, VirtualMachine, VmID, VmSpec};
+
+type Cache = HashMap<VmID, VirtualMachine>;
+
+/// The in-memory cache of virtual machines ([VirtualMachine]) created with Aurae.
+#[derive(Debug)]
+pub struct VirtualMachines {
+    cache: Cache,
+}
+
+impl Default for VirtualMachines {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl VirtualMachines {
+    /// Create a new instance of the virtual machines cache.
+    pub fn new() -> Self {
+        unsafe {
+            let _ = libc::signal(libc::SIGCHLD, libc::SIG_IGN);
+        }
+
+        // Before we start any threads, mask the signals we'll be
+        // installing handlers for, to make sure they only ever run on the
+        // dedicated signal handling thread we'll start in a bit.
+        for sig in &vmm::vm::Vm::HANDLED_SIGNALS {
+            if let Err(e) = block_signal(*sig) {
+                eprintln!("Error blocking signals: {e}");
```

Comment:
should these be fatal errors?  we should also integrate with the logging we have elsewhere (use tracing::{info, trace, warn};)

Reply:
Swapped for tracing in https://github.com/aurae-runtime/aurae/pull/506/commits/1d289d24a98567afff84f634376b020f228b8d4b

> should these be fatal errors?

I'm honestly not sure. The only thing I gather is that the VMM "should" be handling these signals explicitly, but CH [doesn't seem to care](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/9f9a4e657f96cc292ac34354292377a275752819/src/main.rs#L636-L646) if something goes wrong.